### PR TITLE
Add all remaining warlock runes

### PR DESCRIPTION
**Pull request by EmilLindegaard**
Contains 6 warlock runes. (Haunt & Metamorphosis missing as the last two, but they require real quests)

I'm a little unsure whether fake quest 90081 should be discarded and use the real quest instead @Yttrium-tYcLief 

Also a little unsure if I should blacklist the quests that contains many mob icons (Demonic Grace & Master Channeler westfall) ..

Any critique welcome

### CODE REVIEW

**Comment by EmilLindegaard** [comment]
But any critique is selvmord

**Comment by BreakBB** [comment]
If there is a real quest for any step to a rune, then we have to use it and in most cases that also make things easier.

Which quests do you need, I ran the scraper and get hand you the ones, so you can use and correct these.

Also as a first feedback: There is no need to add spaces and comments for individual quest starters. Just make sure they are correctly separated between NPC and object starters and then that's good enough 👍🏻

**Comment by BreakBB** [comment]
Oh and yes: For now hide the runes which otherwise would flood the map or certain zones with `!` because too many mobs are dropping the rune.

**Comment by EmilLindegaard** [comment]
> Which quests do you need, I ran the scraper and get hand you the ones, so you can use and correct these.

For the 6 runes in this PR, no need to add quests.
But for the last two rune Haunt and Metamorphosis the following quests are needed:
- 77621 (for Haunt)
- 77586 (for Haunt)
- 77672 (for Haunt)
- 78681 (for Meta)
- 78684 (for Meta)
- 78702 (for Meta)

> Also as a first feedback: There is no need to add spaces and comments for individual quest starters. Just make sure they are correctly separated between NPC and object starters and then that's good enough 👍🏻

Only did this to make it easier for myself to keep track if I got all the correct mob IDs. I'd keep them, but if you insist, then I'll remove (y)

**Comment by EmilLindegaard** [comment]
fixes incoming this afternoon

**Comment by BreakBB** [comment]
> For the 6 runes in this PR, no need to add quests.

So there are no quests you accept for these runes?

> following quests are needed:

<details>
  <summary>These need to be added to `sodBaseQuests.lua` (correctly sorted by ID)</summary>

```lua
    [77586] = {
        [questKeys.name] = "Stolen Power",
        [questKeys.startedBy] = {{3156}},
        [questKeys.finishedBy] = {{3156,}},
        [questKeys.requiredLevel] = 2,
        [questKeys.questLevel] = 2,
        [questKeys.requiredRaces] = raceIDs.ALL_HORDE,
        [questKeys.requiredClasses] = classIDs.WARLOCK,
        [questKeys.objectivesText] = {"Find the stolen rune inside of the Burning Blade Coven. Learn its power, then report back to Nartok in The Den."},
        [questKeys.objectives] = {nil,nil,nil,nil,nil,{{403919}}},
    },
    [77621] = {
        [questKeys.name] = "Stolen Power",
        [questKeys.startedBy] = {{459}},
        [questKeys.finishedBy] = {{459,}},
        [questKeys.requiredLevel] = 2,
        [questKeys.questLevel] = 2,
        [questKeys.requiredRaces] = raceIDs.ALL_ALLIANCE,
        [questKeys.requiredClasses] = classIDs.WARLOCK,
        [questKeys.objectivesText] = {"Recover the rune from the Defias camps to the east and use it to learn a new ability, then return to Drusilla La Salle in Northshire."},
        [questKeys.objectives] = nil,
    },
    [77672] = {
        [questKeys.name] = "The Lost Rune",
        [questKeys.startedBy] = {{2126}},
        [questKeys.finishedBy] = {{2126,}},
        [questKeys.requiredLevel] = 2,
        [questKeys.questLevel] = 2,
        [questKeys.requiredRaces] = raceIDs.ALL_HORDE,
        [questKeys.requiredClasses] = classIDs.WARLOCK,
        [questKeys.objectivesText] = {"Recover the rune from Night Web's Hollow and use it to learn a new ability, then return to Maximillion in Deathknell."},
        [questKeys.objectives] = {nil,nil,nil,nil,nil,{{403919}}},
    },
    [78681] = {
        [questKeys.name] = "The Conjuring",
        [questKeys.startedBy] = {{6247}},
        [questKeys.finishedBy] = nil,
        [questKeys.requiredLevel] = 20,
        [questKeys.questLevel] = 28,
        [questKeys.requiredRaces] = raceIDs.NONE,
        [questKeys.requiredClasses] = classIDs.NONE,
        [questKeys.objectivesText] = {"Collect 10 Blood of the Legion and perform the ritual to summon Des'Altek to trap its soul once again."},
        [questKeys.objectives] = {nil,nil,{{210966}}},
    },
    [78684] = {
        [questKeys.name] = "Mysterious Traveler",
        [questKeys.startedBy] = {{213444}},
        [questKeys.finishedBy] = {{6247,}},
        [questKeys.requiredLevel] = 20,
        [questKeys.questLevel] = 28,
        [questKeys.requiredRaces] = raceIDs.NONE,
        [questKeys.requiredClasses] = classIDs.NONE,
        [questKeys.objectivesText] = {"Meet with Doan Karhan and the Mysterious Traveler in the Barrens."},
        [questKeys.objectives] = nil,
    },
    [78702] = {
        [questKeys.name] = "Raszel Ander",
        [questKeys.startedBy] = {{6247}},
        [questKeys.finishedBy] = {{6247,}},
        [questKeys.requiredLevel] = 20,
        [questKeys.questLevel] = 28,
        [questKeys.requiredRaces] = raceIDs.NONE,
        [questKeys.requiredClasses] = classIDs.NONE,
        [questKeys.objectivesText] = {"Raszel was a friend... One I thought I had lost long ago, from a different time... We were young apprentices back then... Much has changed.The result of the ritual might have been a setback, but I believe we can still reform the weapon.Raszel and I have much to discuss, together we will come up with a plan. I'll send word once we know how to move forward.In the meantime, take this. It seems like something you might be interested in."},
        [questKeys.objectives] = nil,
    },
```
</details>

Also make sure to fix the objectives for `77586`, `77621` and `77672` in `sodQuestFixes`. These need to get a "spellObjective" and an item connected to it. You can checkout the correction for `77568` to see what I mean.

> Only did this to make it easier for myself to keep track if I got all the correct mob IDs. I'd keep them, but if you insist, then I'll remove :)

That is totally fine and I can understand that. Keep them, if they help you 👍🏻 We know it's quite a big step up to get into the data corrections 😅

**Comment by EmilLindegaard** [comment]
> So there are no quests you accept for these runes?

One of them drops from a mobs during an old classic quest. So no need to add new quests for this one. I'll make this modification this afternoon.

> These need to be added to `sodBaseQuests.lua` (correctly sorted by ID)

But these are added with the wowhead scraper, right? Is it a possibility for me to do this or not?

> Also make sure to fix the objectives for 77586, 77621 and 77672 in sodQuestFixes. These need to get a "spellObjective" and an item connected to it. You can checkout the correction for 77568 to see what I mean.

Will make sure to check this out

**Comment by BreakBB** [comment]
Oh also a small note: I think I started a bad pattern by adding the `--- Mage runes` and then added all kind of IDs below it.

I think we should not do that, so we always know what the next ID is, that we want to add (the one at the bottom). Otherwise (like it is now), we have to look through all IDs to find the next valid one.

> But these are added with the wowhead scraper, right? Is it a possibility for me to do this or not?

Click on the "These need to be added to `sodBaseQuests.lua` (correctly sorted by ID)" text above, to see the data. We scrape that data, but manually add them one by one atm, so we know the quests we add are correct. The wowhead data is quite incomplete and so is the capability of our scraper atm.

**Comment by BreakBB** [comment]
Don't worry about the IDs too much though. We can always fix/change them, once you are done with the relevant stuff 👍🏻

**Comment by BreakBB** [comment]
Whoops, sorry - misclick

**Comment by EmilLindegaard** [comment]
> Click on the "These need to be added to sodBaseQuests.lua (correctly sorted by ID)" text above, to see the data. We scrape that data, but manually add them one by one atm, so we know the quests we add are correct. The wowhead data is quite incomplete and so is the capability of our scraper atm.

Oh I get it now. How did you generate these? Is it something I could have done myself?

**Comment by BreakBB** [comment]
> How did you generate these? Is it something I could have done myself?

That is the result of our wowhead scraper. You can find it [here](https://github.com/Questie/Questie/tree/master/ExternalScripts(DONOTINCLUDEINRELEASE)/scraper).

**Comment by EmilLindegaard** [comment]
Ready for review :)

File: Database/Corrections/sodQuestFixes.lua [hunk 1]
```diff
@@ -42,6 +42,10 @@ function SeasonOfDiscovery:LoadQuests()
             [questKeys.zoneOrSort] = sortKeys.SHAMAN,
             [questKeys.requiredSpell] = -410094,
         },
+        [77586] = {
+            [questKeys.objectives] = {nil, nil, nil, nil, nil, {{403919, nil, 205230}}},
+            [questKeys.requiredSpell] = -403919,
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:47]:
```suggestion
            [questKeys.objectives] = {nil, nil, nil, nil, nil, {{403919, nil, 205230}}},
            [questKeys.zoneOrSort] = sortKeys.WARLOCK,
            [questKeys.requiredSpell] = -403919,
```

File: Database/Corrections/sodQuestFixes.lua [hunk 2]
```diff
@@ -57,6 +61,10 @@ function SeasonOfDiscovery:LoadQuests()
             [questKeys.objectives] = {nil, nil, nil, nil, nil, {{402862, nil, 205951}}},
             [questKeys.requiredSpell] = -402862,
         },
+        [77621] = {
+            [questKeys.objectives] = {nil, nil, nil, nil, nil, {{403919, nil, 205230}}},
+            [questKeys.requiredSpell] = -403919,
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:66]:
```suggestion
            [questKeys.objectives] = {nil, nil, nil, nil, nil, {{403919, nil, 205230}}},
            [questKeys.zoneOrSort] = sortKeys.WARLOCK,
            [questKeys.requiredSpell] = -403919,
```

File: Database/Corrections/sodQuestFixes.lua [hunk 3]
```diff
@@ -90,6 +98,10 @@ function SeasonOfDiscovery:LoadQuests()
             [questKeys.objectives] = {nil, nil, nil, nil, nil, {{402862, nil, 205951}}},
             [questKeys.requiredSpell] = -402862,
         },
+        [77672] = {
+            [questKeys.objectives] = {nil, nil, nil, nil, nil, {{403919, nil, 205230}}},
+            [questKeys.requiredSpell] = -403919,
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:103]:
```suggestion
            [questKeys.objectives] = {nil, nil, nil, nil, nil, {{403919, nil, 205230}}},
            [questKeys.zoneOrSort] = sortKeys.WARLOCK,
            [questKeys.requiredSpell] = -403919,
```

File: Database/Corrections/sodQuestFixes.lua [hunk 4]
```diff
@@ -1129,6 +1173,250 @@ function SeasonOfDiscovery:LoadQuests()
...
+            [questKeys.startedBy] = {{202060}},
+            [questKeys.finishedBy] = nil,
+            [questKeys.requiredLevel] = 1,
+            [questKeys.questLevel] = 9,
+            [questKeys.requiredRaces] = raceIDs.ALL_HORDE,
+            [questKeys.requiredClasses] = classIDs.WARLOCK,
+            [questKeys.objectivesText] = {"Gather some Mage and Warlock friends and attack the Frozen Murloc with fire spells."},
+            [questKeys.requiredSpell] = -403925,
+            [questKeys.zoneOrSort] = sortKeys.WARLOCK,
+        },
```
Comment (BreakBB) [inline Database/Corrections/sodQuestFixes.lua:1211]: These can be combined into a single one and simply make the `startedBy` contain both IDs. Like so:

```lua
            [questKeys.startedBy] = {{208275,202060}},
```

**Comment by BreakBB** [comment]
I had a short look and already added some missing `zoneOrSort`, nothing big.

These look really really great! You did a great job on such a task 🥳 

Do you feel confortable with resolving the conflicts?

File: Database/Corrections/sodQuestFixes.lua (hunk 4 above)
Comment (EmilLindegaard) [reply Database/Corrections/sodQuestFixes.lua:1226]: Even thought the creature is different?
Would you then prefer the objective text to state "Frozen Makrura/Murloc"?

**Comment by EmilLindegaard** [comment]
> I had a short look and already added some missing `zoneOrSort`, nothing big.
> 
> These look really really great! You did a great job on such a task 🥳
> 
> Do you feel confortable with resolving the conflicts?

I usually resolve merge conflics locally by rebasing onto your branch and then pushing. But I possibly had to do a force push in that case and I'm not aware if that makes the PR commit history messy.

I used github for merging this time, but not sure if I really like it tbh :)

File: Database/Corrections/sodQuestFixes.lua (hunk 4 above)
Comment (BreakBB) [reply Database/Corrections/sodQuestFixes.lua:1226]: Ah. I missed that one. Then this is totally fine, thank you :D

**Review by BreakBB — APPROVED** [review]
Looks fine to me 👍🏻

**Pull request merged**